src/general/StageBase.cs

```diff
@@ -25,6 +25,9 @@ public abstract class StageBase<TPlayer> : NodeWithInput, IStage, IGodotEarlyNod
     protected PauseMenu pauseMenu = null!;
     protected Control hudRoot = null!;
 
+    [JsonProperty]
+    protected DayNightCycle? lightCycle;
```

Comment:
The intention seems that the light cycle is never null, so this should not be nullable and use null suppression instead.

Reply:
As far as I can tell this needs to be here otherwise jetbrains gets mad at this check:

```
    public override void _Ready()
    {
        base._Ready();

        // Need to check for null light cycle in case this is loaded from an old save
        if (!IsLoadedFromSave || lightCycle == null)
        {
            lightCycle = new DayNightCycle(
                GameWorld.WorldSettings.DayNightCycleEnabled,
                GameWorld.WorldSettings.DayLength);
        }
    }
```

After that check it should be impossible for lightCycle to be null

Reply:
You can write the null check by putting the nullability override there, at least I'm pretty sure you can do that (`lightCycle == null!`).